### ISSUE TITLE
Add support for usersync

### ISSUE THREAD

**Issue by junghans** [hasufell/portage-gentoo-git-config#16]
`FEATURES="usersync"` drops privileges to the owner of `${repository_location}`, however scripts in `repo.postsync.d` are still run as root, which leads to strange effects like metadata being owned by root and not the developer or rsync being run as root.

**Comment by hasufell** [comment]
I think the easiest thing is just to do:

``` sh
sudo -u <user> -g <group> egencache --jobs=${egencache_jobnum} --repo=${TARGET} --update --update-use-local-desc
```

But then I don't know if automagic user/group detection makes sense.

**Comment by redneb** [comment]
Isn't invoking the scripts as the right user a responsibility for portage itself?

**Comment by hasufell** [comment]
@dol-sen

**Comment by junghans** [comment]
We could put the `<user>` and `<group>` as a variable into `make.conf`, but I guess portage should run the post-sync hooks with the same rights are the sync itselfs

**Comment by dol-sen** [comment]
make.conf is the wrong place for that.  They are repo specific, so need to be in the correct repos.conf/ file.

There we will need to change the code to use the repo's user and group settings for the postsync hooks

**Comment by junghans** [comment]
Yeah, `repos.conf` seems like a better place, but why not auto-detecting it like the `usersync` feature does it?

**Comment by dol-sen** [comment]
We have started discussing the ways to accomplish it.  I believe the default for usersync is to use the current user info.  You were mentioning setting the user info.  That would have to be done via a per repo setting like I mentioned.  But for a start, a simple boolean in the repo's  config section could be used to toggle normal usersync permissions for that repo.

Also, Could you please open a bug in bugzilla for this, they don't have github/bugzilla set up to be capable of our bugzilla mirroring things discussed here.  So there is a possibility of things here getting lost/forgotten.  FYI, I don't normally follow most repos/notifications from github.  I already get more email than I can devote time to read, etc...    I was gone for work for 2 weeks, had 2K+ emails to clean out of my gentoo inbox.

**Comment by junghans** [comment]
Report on bugzilla as [bug #565718](https://bugs.gentoo.org/show_bug.cgi?id=565718)

**Comment by hasufell** [comment]
I don't really use gentoo anymore. Either use a fork or I can give you push access to this repo.

**Comment by pavlix** [comment]
@hasufell I recently started using this project not being aware that you're no longer maintaining it. @junghans What's your position on that?

**Comment by junghans** [comment]
I am still using it, but @srcshelton has a done some development recently.

@hasufell: could you transfer the ownership of his repo to @gentoo?
